### ISSUE TITLE
Inconsistent keras.layers.normalization version

### ISSUE THREAD

**Issue by tiok-cek1** [tensorflow/tensorflow#43871]
**System information**
- Have I written custom code: Yes
- OS Platform and Distribution: Docker image python 3.8 (debian distro) & Google Colab
- TensorFlow installed from pip
- TensorFlow version 2.3.0
- Python version: docker python 3.8.2 & Colab python 3.6.9

**Describe the current behavior**
When using `BatchNormalization `from different keras tree got a different result. In my case the model failed to converge.
`tensorflow.keras` & `tensorflow.python.keras` both using the same version of keras but if we dump the `BatchNormalization `layer the version is different. 

**Describe the expected behavior**
The same version of keras, must use the same version of normalization.

**Standalone code to reproduce the issue**
```
import sys

import tensorflow

from tensorflow import keras as k1
from tensorflow.python import keras as k2

from tensorflow.keras.layers import BatchNormalization as BN1
from tensorflow.python.keras.layers import BatchNormalization as BN2

print('python version:', sys.version)
print('tf version:', tensorflow.__version__)
print('keras version:', k1.__version__)
print('python.keras version:', k2.__version__)

batch_normalization_layer_keras = BN1()
print('keras:', batch_normalization_layer_keras)
batch_normalization_layer_python_keras = BN2()
print('python.keras:', batch_normalization_layer_python_keras)
```

**Output**
```
python version: 3.6.9 (default, Jul 17 2020, 12:50:27) 
[GCC 8.4.0]
tf version: 2.3.0
keras version: 2.4.0
python.keras version: 2.4.0
keras: <tensorflow.python.keras.layers.normalization_v2.BatchNormalization object at 0x7f2aba63f390>
python.keras: <tensorflow.python.keras.layers.normalization.BatchNormalization object at 0x7f2aba63fb70>
```

**Comment by ravikyram** [comment]
I have tried in colab with TF version 2.3,nightly version(`2.4.0-dev20201007`) and was able to reproduce the issue.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/0652ca81a568ec8ec31c0f7fb59da8b9/untitled434.ipynb).Thanks!

**Comment by jvishnuvardhan** [comment]
@tiok-cek1 Anything under `tf.python.*` is private, intended for development only, rather than public use. Importing from tensorflow.python or any other modules (including import tensorflow_core...) is not supported, and can break unannounced. For full response on similar issue, you can check here. https://github.com/tensorflow/tensorflow/issues/33075

**Comment by tiok-cek1** [comment]
I opened tf1 and tf2 documentation, both provide the same default parameters. No information about v1 and v2 normalization, only know to use v1 must set [_USE_V2_BEHAVIOR](https://github.com/tensorflow/tensorflow/blob/v2.3.0/tensorflow/python/keras/layers/normalization.py#L147-L149) to False.
I ran some code from internet, the optimization failed to converge because of this version changes (code can run without error but produce a different result).

I think this issue is done, thanks @jvishnuvardhan for explaining the difference of `tf.python.keras` and `tf.keras`.
